src/plugins/interactive_setup/public/cluster_configuration_form.tsx

```diff
@@ -258,9 +258,9 @@ export const ClusterConfigurationForm: FunctionComponent<ClusterConfigurationFor
               value="true"
               checked={!!form.values.caCert}
               onChange={() => {
-                const intermediateCa = certificateChain[Math.min(1, certificateChain.length - 1)];
+                const rootCa = certificateChain[certificateChain.length - 1];
```

Comment:
Looking at this, it occurred to me that the last certificate in the chain might not always be the root CA.
I did some digging and found that OpenSSL does not always return the correct order of the chain (EE to root), but Node's `getPeerCertificate` API does contain logic that ensures the order of the cert chain is correct:

1. https://github.com/nodejs/node/blob/ea47d836403fa055861b01ea791f9b2da7ee7e7e/lib/_tls_wrap.js#L987
2. https://github.com/nodejs/node/blob/79d681501d592eceaef104c8ed43856158d8cf1e/src/crypto/crypto_tls.cc#L1579
3. https://github.com/nodejs/node/blob/79d681501d592eceaef104c8ed43856158d8cf1e/src/crypto/crypto_common.cc#L955
4. https://github.com/nodejs/node/blob/79d681501d592eceaef104c8ed43856158d8cf1e/src/crypto/crypto_common.cc#L429-L463

So, I think this is the correct fix :+1:

Reply:
Ah interesting. We actually manually traverse the certificate chain anyways so this is guaranteed to begin order: https://github.com/thomheymann/kibana/blob/interactive-setup-root-ca/src/plugins/interactive_setup/server/elasticsearch_service.ts#L369